java/cdk/src/main/java/sleeper/cdk/stack/core/AutoStopEcsClusterTasksStack.java

```diff
@@ -36,27 +41,33 @@
 import java.util.Map;
 import java.util.Objects;
 
+/**
+ * Stops ECS Cluster tasks for the CloudFormation stack.
+ */
 @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
-public class AutoStopEcsClusterTasks {
+public class AutoStopEcsClusterTasksStack extends NestedStack {
 
-    private AutoStopEcsClusterTasks() {
-    }
+    private final IFunction lambda;
+    private final Provider provider;
+
+    public AutoStopEcsClusterTasksStack(Construct scope, String id, InstanceProperties instanceProperties, BuiltJars jars,
+            LoggingStack loggingStack) {
 
-    public static void autoStopTasksOnEcsCluster(
-            Construct scope, InstanceProperties instanceProperties, LambdaCode lambdaCode,
-            ICluster cluster, String clusterName,
-            ILogGroup logGroup,
-            ILogGroup providerLogGroup) {
+        super(scope, id);
 
-        String id = cluster.getNode().getId() + "-AutoStop";
-        String functionName = clusterName + "-autostop";
+        // Jars bucket
+        IBucket jarsBucket = Bucket.fromBucketName(this, "JarsBucket", jars.bucketName());
+        LambdaCode lambdaCode = jars.lambdaCode(jarsBucket);
 
-        IFunction lambda = lambdaCode.buildFunction(scope, LambdaHandler.AUTO_STOP_ECS_CLUSTER_TASKS, id + "Lambda", builder -> builder
+        String functionName = String.join("-", "sleeper",
+                Utils.cleanInstanceId(instanceProperties), "auto-stop-ecs-cluster-tasks");
+
+        lambda = lambdaCode.buildFunction(this, LambdaHandler.AUTO_STOP_ECS_CLUSTER_TASKS, "Lambda", builder -> builder
                 .functionName(functionName)
                 .memorySize(2048)
                 .environment(EnvironmentUtils.createDefaultEnvironmentNoConfigBucket(instanceProperties))
                 .description("Lambda for auto-stopping ECS tasks")
-                .logGroup(logGroup)
+                .logGroup(loggingStack.getLogGroup(LogGroupRef.AUTO_STOP_ECS_CLUSTER_TASKS))
                 .timeout(Duration.minutes(10)));
```

Comment:
This should be 15 minutes (the maximum possible) to maximise the chances of deleting everything before the lambda times out (I know it wasn't changed here but we should change it in this PR anyway).